pkg/util/k8sutil/k8sutil.go

```diff
@@ -290,15 +292,31 @@ func NewEtcdPod(m *etcdutil.Member, initialCluster []string, clusterName, state,
 }
 
 func MustNewKubeClient() kubernetes.Interface {
-	cfg, err := rest.InClusterConfig()
+	cfg, err := InClusterConfig()
 	if err != nil {
 		panic(err)
 	}
 	return kubernetes.NewForConfigOrDie(cfg)
 }
 
+func InClusterConfig() (*rest.Config, error) {
+	// Work around https://github.com/kubernetes/kubernetes/issues/40973
+	// See https://github.com/coreos/etcd-operator/issues/731#issuecomment-283804819
+	if len(os.Getenv("KUBERNETES_SERVICE_HOST")) == 0 {
+		addrs, err := net.LookupHost("kubernetes.default.svc")
```

Comment:
Let's keep it simple atm. We can find out if retry is needed once more testing happens.